tests/integration/pilot/vm/vm_test.go

```diff
@@ -15,77 +15,62 @@
 package vm
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
+	"istio.io/istio/pkg/test/framework/label"
+
+	"istio.io/istio/pkg/test/framework/components/namespace"
+
 	"istio.io/istio/pkg/config/protocol"
 	"istio.io/istio/pkg/test/framework"
 	"istio.io/istio/pkg/test/framework/components/echo"
 	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
-	"istio.io/istio/pkg/test/framework/components/istio"
-	"istio.io/istio/pkg/test/framework/components/namespace"
-	"istio.io/istio/pkg/test/framework/components/pilot"
-	"istio.io/istio/pkg/test/framework/resource"
 	"istio.io/istio/pkg/test/util/retry"
 )
 
-var p pilot.Instance
-var ns namespace.Instance
+func TestVmOS(t *testing.T) {
+	vmImages := []string{"app_sidecar_ubuntu_bionic"}
+	VMTestBody(t, vmImages)
+}
 
-// This tests VM mesh expansion. Rather than deal with the infra to get a real VM, we will use a pod
-// with no Service, no DNS, no service account, etc to simulate a VM.
-func TestMain(m *testing.M) {
-	framework.
-		NewSuite(m).
-		RequireSingleCluster().
-		Setup(func(ctx resource.Context) error {
-			var err error
-			ns, err = namespace.New(ctx, namespace.Config{
-				Prefix: "virtual-machine",
-				Inject: true,
-			})
-			return err
-		}).
-		Setup(istio.Setup(nil, func(cfg *istio.Config) {
-			cfg.ControlPlaneValues = `
-values:
-  global:
-    meshExpansion:
-      enabled: true`
-		})).
-		Setup(func(ctx resource.Context) (err error) {
-			if p, err = pilot.New(ctx, pilot.Config{}); err != nil {
-				return err
-			}
-			return nil
-		}).
-		Run()
+func TestVmOSPost(t *testing.T) {
+	vmImages := []string{"app_sidecar_ubuntu_xenial", "app_sidecar_ubuntu_focal", "app_sidecar_ubuntu_bionic",
+		"app_sidecar_debian_9", "app_sidecar_debian_10"}
+	VMTestBody(t, vmImages, label.Postsubmit)
 }
 
-func TestVmTraffic(t *testing.T) {
+func VMTestBody(t *testing.T, vmImages []string, label ...label.Instance) {
 	framework.
 		NewTest(t).
+		Features("traffic.reachability").
+		Label(label...).
 		Run(func(ctx framework.TestContext) {
+			ns = namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "virtual-machine",
+				Inject: true,
+			})
 			// Set up strict mTLS. This gives a bit more assurance the calls are actually going through envoy,
 			// and certs are set up correctly.
 			ctx.ApplyConfigOrFail(ctx, ns.Name(), `
 apiVersion: security.istio.io/v1beta1
 kind: PeerAuthentication
 metadata:
-  name: default
+ name: default
```

Comment:
does this work? looks like its not enough spaces... or does it not matter?

Reply:
Well the test passes so I guess it does :slightly_smiling_face:

Reply:
I am not sure. I don't think I explicitly change these spaces other than running `make format`. I will do some experiments and let you know.